config/config.go

```diff
@@ -27,6 +28,12 @@ type Config struct {
 	ShutdownTimeOut int                 `yaml:"shutdowntimeout"`
 }
 
+type MetricsConfig struct {
+	Enabled bool   `yaml:"enabled"`
```

Comment:
still thinking about this but i wonder if instead of `enabled` we might implicitly enable metrics if either prefix or port are set?   

if you like that idea, we'd want a default value for port, for example `9199`

partly suggesting this approach because i see an example below that makes no sense (harmless though, so it's not a big deal) why bother configuring prefix/port if metrics are not exposed

```
metrics:
  enabled: false
  port: 9877
  prefix: logging
  ```

Reply:
enabled: false would allow me to turn on and off metrics, previously the code base would only turn off metrics if the port was 0. I thought the enabled was more clearly stated in the configuration.